### PR TITLE
Added test for time jumps

### PR DESCRIPTION
**Pull request by rmstar**

### CODE REVIEW

File: test/cpp/common/time_jump_test.cc [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    grpc_core::MutexLock lock(&mu);
+    std::thread thd = std::thread([]() {
+      gpr_sleep_until(grpc_timeout_milliseconds_to_deadline(100));
+      std::ostringstream cmd;
+      cmd << "sudo date `date -v" << GetParam() << " \"+%m%d%H%M%y\"`";
+      run_cmd(cmd.str().c_str());
+    });
+    gpr_timespec before = gpr_now(GPR_CLOCK_MONOTONIC);
+    int timedout =
+        cond.Wait(&mu, grpc_millis_to_timespec(1500, GPR_CLOCK_MONOTONIC));
```
[the remark this answers is not in the document]
Comment (rmstar) [reply test/cpp/common/time_jump_test.cc]: Fixed.

File: tools/internal_ci/macos/grpc_time_tests.cfg [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2019 The gRPC Authors
```
[the remark this answers is not in the document]
Comment (rmstar) [reply tools/internal_ci/macos/grpc_time_tests.cfg]: Removed new test job and added to `tools/internal_ci/macos/grpc_run_bazel_tests.sh`

File: test/cpp/common/time_jump_test.cc [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  grpc_timer timer;
+  grpc_timer_init(&timer, grpc_core::ExecCtx::Get()->Now() + 3000,
+                  GRPC_CLOSURE_CREATE(
+                      [](void*, grpc_error* error) {
+                        GPR_ASSERT(error == GRPC_ERROR_CANCELLED);
+                      },
+                      nullptr, grpc_schedule_on_exec_ctx));
+  gpr_sleep_until(grpc_timeout_milliseconds_to_deadline(100));
+  std::ostringstream cmd;
+  cmd << "sudo date `date -v" << GetParam() << " \"+%m%d%H%M%y\"`";
```
[the remark this answers is not in the document]
Comment (rmstar) [reply test/cpp/common/time_jump_test.cc]: I was not able to build timer tests in the old code, maybe some incompatibility with gcc version?
```
./src/core/lib/transport/static_metadata.h:244:14: error: unknown type name 'grpc_slice_refcount_vtable'; did you mean 'grpc_slice_refcount'?
extern const grpc_slice_refcount_vtable grpc_static_metadata_vtable;
             ^~~~~~~~~~~~~~~~~~~~~~~~~~
             grpc_slice_refcount
/usr/local/include/grpc/impl/codegen/slice.h:50:8: note: 'grpc_slice_refcount' declared here
```

[the remark this answers is not in the document]
Comment (muxi) [reply test/cpp/common/time_jump_test.cc]: That's strange error. I'm not going to push hard on this though.

**Pull request merged**